src/JWT/JwtValidator.cs

```diff
@@ -37,14 +37,19 @@ public void Validate(string payloadJson, string decodedCrypto, string decodedSig
             object expObj;
             if (payloadData.TryGetValue("exp", out expObj))
             {
-                int expInt;
+                if (expObj == null)
+                {
+                    throw new SignatureVerificationException("Claim 'exp' must be a double.");
+                }
+
+                double expInt;
                 try
                 {
-                    expInt = Convert.ToInt32(expObj);
+                    expInt = Convert.ToDouble(expObj);
                 }
                 catch
                 {
-                    throw new SignatureVerificationException("Claim 'exp' must be an integer.");
+                    throw new SignatureVerificationException("Claim 'exp' must be a double.");

```

Comment:
I think "Claim 'exp' must be a number." would remove some potential confusion from the error message.